### PR TITLE
Develop donation/reimbursement

### PR DESCRIPTION
**Pull request by pianote**
1. [VolunteerDetail]: make donation before reimbursement created
2. [MyVolunteerDetail]: able to skip reimbursement

### CODE REVIEW

File: requests/serializers.py [hunk 1]
```diff
@@ -58,14 +59,20 @@ def get_reimbursement(self, obj):
...
-            "reimbursement_id": reimbursement.id}
+            return ReimbursementSerializer(reimbursement).data
         except:
             return None
-    
+
+    def get_delivery_status(self, obj):
+        try:
+            volunteer = Volunteer.objects.get(request=obj)
+            if volunteer.status == 'Delivered':
```
Comment (bri616) [inline requests/serializers.py:69]: Sometimes in big projects I have found it nice to store strings like this in constants so we can reference a variable name everywhere. That way, if the string needs to change, it only needs to change in one place.

So maybe in `requests/request_form_choices.py` define:

```
VOLUNTEERING_STATUS_SIGNED_UP = 'Signed Up'
VOLUNTEERING_STATUS_READY = 'Ready'
VOLUNTEERING_STATUS_DELIVERED = 'Delivered'

VOLUNTEERING_STATUS_CHOICES = (
    (VOLUNTEERING_STATUS_SIGNED_UP, 'Signed Up'),
    (VOLUNTEERING_STATUS_READY, 'Ready'),
    (VOLUNTEERING_STATUS_DELIVERED, 'Delivered')
)
```

and then the logic in here can reference the constant like this? This means everywhere you reference any `volunteer.status` you can use the constants! What do you think about this kind of pattern?

```
return volunteer.status if volunteer.status == VOLUNTEER_STATUS_DELIVERED else 'On the way'
```

File: users/migrations/0002_useraccount_display_name.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.0.7 on 2020-08-03 23:39
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('users', '0001_initial'),
```
Comment (bri616) [inline users/migrations/0002_useraccount_display_name.py:9]: is the initial `users` migration missing? I don't see it in this PR

**Review by bri616 — APPROVED** [review]
I read through this and left some notes, hopefully they made sense! I noticed the initial migration for `users` is missing. If you don't explicitly have it, I think that's probably ok -- we can basically squash all of the `users` migrations if we want into one initial one.

**Comment by pianote** [comment]
@bri616 Your comments are so helpful Bri. I appreciate it so much.

File: requests/serializers.py (hunk 1 above)
Comment (aphunk) [reply requests/serializers.py:69]: +1 for constants!

**Review by aphunk — APPROVED** [review]
This looks good, and pretty much +1 to all of Bri's comments.

File: .gitignore [hunk 3]
```diff
@@ -7,7 +7,6 @@ maskes-ui/
 backup/
 venv/
 __pycache__/
-migrations/
```
[the remark this answers is not in the document]
Comment (pianote) [reply .gitignore:10]: That's exactly what I did, manually run migrate on server. :grimacing:

File: users/migrations/0002_useraccount_display_name.py (hunk 2 above)
Comment (pianote) [reply users/migrations/0002_useraccount_display_name.py:9]: found my users 0001_initial in an old branch! don't know why I lost it :grin:

File: requests/serializers.py [hunk 4]
```diff
@@ -58,14 +59,20 @@ def get_reimbursement(self, obj):
...
-            "reimbursement_amount": reimbursement.amount,
-            "reimbursement_id": reimbursement.id}
+            return ReimbursementSerializer(reimbursement).data
         except:
             return None
-    
+
+    def get_delivery_status(self, obj):
+        try:
+            volunteer = Volunteer.objects.get(request=obj)
```
[the remark this answers is not in the document]
Comment (pianote) [reply requests/serializers.py:68]: what if we don't want a 404 response. 
instead return a null value for delivery_status when the volunteer object doesn't exist
How about avoid try/except block by checking 
`Volunteer.objects.filter(request=obj).exists()`

File: requests/serializers.py (hunk 1 above)
Comment (pianote) [reply requests/serializers.py:69]: This is so great. I made the change! 
Thanks a lot bri :pray:

**Pull request merged**